src/components/Pagination/Pagination.tsx

```diff
@@ -385,307 +415,625 @@ class Pagination extends React.PureComponent<IProps, IState> {
-          showTitle={props.showTitle}
-          itemRender={props.itemRender}
+          key={i}
+          page={i}
+          active={active}
+          showTitle={showTitle === true}
+          itemRender={itemRender}
+          componentStyle={{ transition: 'none' }}
         />
       );
+    }
+  }
 
-      let left = Math.max(1, current - pageBufferSize);
-      let right = Math.min(current + pageBufferSize, allPages);
+  renderSimplePagination = () => {
+    const { simplePagination, pageSize, theme } = this.props;
 
-      if (current - 1 <= pageBufferSize) {
-        right = 1 + pageBufferSize * 2;
-      }
+    const { initialPageSize } = this.state;
 
-      if (allPages - current <= pageBufferSize) {
-        left = allPages - pageBufferSize * 2;
-      }
+    const numPages = calculatePage(undefined, this.state, this.props);
 
-      for (let i = left; i <= right; i++) {
-        const active = current === i;
-        pagerList.push(
-          <Pager
-            locale={props.locale}
-            rootPrefixCls={theme['rc-pagination']}
-            onClick={this.handleChange}
-            onKeyPress={this.runIfEnter}
-            key={i}
-            page={i}
-            active={active}
-            showTitle={props.showTitle}
-            itemRender={props.itemRender}
-          />
-        );
-      }
+    if (simplePagination && numPages > 1) {
+      return (
+        <li className={theme['rc-pagination-simple-pagination']}>
+          <Button onClick={this.loadMore}>Show More Results</Button>
+        </li>
+      );
+    }
+    if (simplePagination && numPages === 1 && initialPageSize !== pageSize) {
+      return (
+        <li className={theme['rc-pagination-simple-pagination']}>
+          <Button
+            disabled={initialPageSize !== pageSize && numPages === 1}
+            onClick={this.loadMore}
+          >
+            Show More Results
+          </Button>
+        </li>
+      );
+    }
+    return null;
+  }
 
-      if (current - 1 >= pageBufferSize * 2 && current !== 1 + 2) {
-        pagerList[0] = React.cloneElement(pagerList[0], {
-          className: theme['rc-pagination-item-after-jump-prev'],
-        });
-        pagerList.unshift(jumpPrev);
-      }
-      if (allPages - current >= pageBufferSize * 2 && current !== allPages - 2) {
-        pagerList[pagerList.length - 1] = React.cloneElement(pagerList[pagerList.length - 1], {
-          className: theme['rc-pagination-item-before-jump-next'],
-        });
-        pagerList.push(jumpNext);
-      }
+  renderJumpToPage = () => {
 
-      if (left !== 1) {
-        pagerList.unshift(firstPager);
-      }
-      if (right !== allPages) {
-        pagerList.push(lastPager);
-      }
+    const {
+      theme,
+      showLessItems,
+    } = this.props;
+
+    const { current = 1, pageSize } = this.state;
+
+    const numPages = calculatePage(undefined, this.state, this.props);
+
+    const maxPageLimit = showLessItems ? 7 : 9;
```

Comment:
how did we come to know numbers like 7 & 9

Reply:
We can make the maximum allowed Pager Buttons to 7. I think there is no need of 9 Buttons. The 9 button Pagination is by default enabled. Currently we disable it by setting the `'showLessItems' props to 'true'`

<img width="446" alt="Screenshot 2021-04-04 at 4 51 27 PM" src="https://user-images.githubusercontent.com/79473218/113508080-a8257b00-956b-11eb-994f-a94a687f08bf.png">

[https://material-ui.com/components/pagination/](https://material-ui.com/components/pagination/)


`[PREV]` ` [1]`  `[2]`  `[3]`  `[4]`  `[5]` ` [...]`  `[10]`  `[NEXT]`   |---> Selected Page 1, Total Page 10
`[PREV]`  `[1]`  `[...]`  `[5]`  `[6]`  `[7]`  `[...]`  `[10]`  `[NEXT]`  |---> Selected Page 6, Total Page 10
`[PREV]`  `[1]`  `[...]`  `[6]`  `[7]`  `[8]`  `[9]`  `[10]`  `[NEXT]`   |---> Selected Page 10, Total Page 10
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[5]`  `[6]`  `[7]`  `[NEXT]`      |---> Selected Page 4, Total Page 7
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[5]`  `[6]`  `[NEXT]`                |---> Selected Page 5, Total Page 6
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[5]`  `[NEXT]`                           |---> Selected Page 2, Total Page 5
`[PREV]`  `[1]`  `[2]`  `[3]`  `[4]`  `[NEXT]`                                     |---> Selected Page 2, Total Page 4
`[PREV]`  `[1]`  `[2]`  `[3]`  `[NEXT]`                                                |---> Selected Page 2, Total Page 3